src/routes/groups/groups.router.js

```diff
@@ -0,0 +1,87 @@
+const express = require('express');
+const { listGroup, postGroup, updateGroup, deleteGroup } = require("./groups.controller");
+const { check, param } = require('express-validator');
+const   { validateBody } = require('../../middleware/validate-body');
+
+const router = express.Router();
+/**
+ * @swagger
+ * /groups:
+ *   get:
+ *     description: Retreive a list of community groups
+ *     produces:
+ *      - application/json
+ *     responses:
+ *       200:
+ *         description: List of community groups successfully retrieved
+ *         schema:
+ *           type: array
+ */
+router.get("", listGroup);
+/**
+ * @swagger
+ * /groups:
+ *   post:
+ *     description: Creates a new community group entry
+ *     produces:
+ *      - application/json
+ *     responses:
+ *       201:
+ *         description: Community group entry has been created
+ */
+
+router.post("", [
+  check("id")
+  .not()
+  .isEmpty()
+  .withMessage('must have id field'),
+  check("id")
+  .not()
+  .isString()
+  .withMessage('must have id as type integer'),
+], validateBody, postGroup);
```

Comment:
I'm surprised to see this validation logic in the router.  Where do you think would be a better place for it?